department-of-veterans-affairs/abd-vro
# EP Merge: Add custom metrics

## What was the problem?

Currently custom metrics are not logged in EP Merge, and the only datadog metrics that can be surfaced are based off log outputs. 

Associated tickets or Slack threads:
- #2055 

## How does this fix it?[^1]

This test uses the datadog python library to add custom metrics. See this [comment](https://github.com/department-of-veterans-affairs/abd-vro/issues/2055#issuecomment-1922081295) for more info on the metrics that should be logged. 

## How to test this PR
Note: the metrics added to this PR cannot be exercised locally, but can only be exercised in a deployed environment with access to the initialization variables for LHDI datadog agent. Testing steps below are merely to ensure that adding the metrics did not break anything, and that the call to log the metrics are made when appropriate. 

- pull code
- Unit Test:
  - from /domain-ee/ee-ep-merge-app run pytest
- Integration Tests:
  - run [base platform](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Docker-Compose#platform-base)
  - from /domain-ee/ee-ep-merge-app run pytest ./integration
- End to End tests:
  - run this [CI](https://github.com/department-of-veterans-affairs/abd-vro/actions/workflows/ee-ep-merge-end-to-end.yml) action to verify successful integration test
  - run end to end tests locally:
    - `export BIP_CLAIM_URL=mock-bip-claims-api:20300`
    - run bgs and bip services and mocks:
      - `COMPOSE_PROFILES="bip,bgs" ./gradlew :app:dockerComposeUp`
      - `COMPOSE_PROFILES="bip,bgs" ./gradlew -p mocks :dockerComposeUp`
    - `./gradlew :domain-ee:ee-ep-merge-app:endToEndTest`


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
